wdl/transforms/new-base/src/main/scala/wdl/transforms/base/wdlom2wom/WorkflowDefinitionElementToWomWorkflowDefinition.scala

```diff
@@ -87,10 +88,12 @@ object WorkflowDefinitionElementToWomWorkflowDefinition {
 
       currentValidation flatMap { currentList =>
         val availableValues: Map[String, OutputPort] = (for {
+          // Anonymous expression nodes are one-time-use and do not exist in the universe of available linking candidates (#3999)
+          node <- currentList.filterNot(_.isInstanceOf[AnonymousExpressionNode])
           node <- currentList
           port <- node.outputPorts
         } yield outputName(node, port) -> port).toMap
-        val nextGraphNodeValidation = WorkflowGraphElementToGraphNode.convert(GraphNodeMakerInputs(next, linkedGraph.consumedValueLookup, availableValues, linkedGraph.typeAliases, workflowName, insideAScatter, callables))
+        val nextGraphNodeValidation: ErrorOr[Set[GraphNode]] = WorkflowGraphElementToGraphNode.convert(GraphNodeMakerInputs(next, linkedGraph.consumedValueLookup, availableValues, linkedGraph.typeAliases, workflowName, insideAScatter, callables))
```

Comment:
This made me notice that the name is a little confusing... maybe while you're here we could update it to something less obtuse, eg `generatedGraphNodesValidation`?

Reply:
Yes, that is exactly why I added the annotation!

I agree with pluralizing the name as well